src/VariationPriceHandler.php

```diff
@@ -0,0 +1,211 @@
+
+use Drupal\commerce_product\Entity\ProductVariation;
+use Drupal\commerce_price\Price;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\commerce_product\ProductAttributeFieldManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\commerce_attributes\Entity\ProductAttributeValue;
+use Drupal\commerce_attributes\Entity\ProductAttribute;
+
+/**
+ * Handles the Variation price calculations.
+ */
+class VariationPriceHandler {
+
+  /**
+   * Logger Service.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $Logger;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * The attribute field manager.
+   *
+   * @var \Drupal\commerce_product\ProductAttributeFieldManagerInterface
+   */
+  protected $attributeFieldManager;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(LoggerChannelFactoryInterface $logger_factory, EntityManagerInterface $entity_manager, EntityTypeManagerInterface $entity_type_manager, ProductAttributeFieldManagerInterface $attribute_field_manager) {
+    $this->logger = $logger_factory->get('system');
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityManager = $entity_manager;
+    $this->attributeFieldManager = $attribute_field_manager;
+  }
+
+  /**
+   * Retrieves the product variations.
+   *
+   * @param array $types
+   *   The product variation types.
+   * @param string $attribute
+   *   The attribute machine name.
+   * @param int $attributeValue
+   *   The Attribue value id.
+   *
+   * @return array
+   *   Array of variations id.
+   */
+  public function getProductVariation(array $types, $attribute, $attributeValue) {
+    // The attribute field name of variation type will always be in the
+    // format of `attrubute_attribute_machine_name`.
+    $attributefieldName = 'attribute_' . $attribute;
+
+    // Get all variations associated with an particular attribute value.
+    $query = \Drupal::entityQuery('commerce_product_variation')
+      ->condition('type', $types, 'IN')
+      ->condition($attributefieldName, $attributeValue);
+    $variations = $query->execute();
+    return $variations;
+  }
+
+  /**
+   * Recalculates the product variation price based on the attribute price.
+   *
+   * @param Drupal\commerce_attributes\Entity\ProductAttributeValue $attributeValue
+   *   The ProductAttributeValue entity.
+   */
+  public function updateProductVariationsPrice(ProductAttributeValue $attributeValue) {
+    $attribute = $attributeValue->getAttribute();
+
+    // Get all variation type associated with the attribute
+    // which is being updated.
+    $variationTypes = $this->getAllVariationTypes($attribute);
```

Comment:
**CHANGE REQUEST**

All variables should be `snake_case` e.g. `$attribute_value`, `$variation_types`.

Reply:
Done